### PR TITLE
feat: add dashboard chart

### PR DESCRIPTION
**Pull request by pelzman**
**Description of PR**

added dashboard chart
it analyzes the monthly income of the User
  link [CU-86bwmcmbc](https://app.clickup.com/t/86bwmcmbc)
---

Please check all that apply.

#### Checklist:

- [x] Rebased with master
- [x] PR Description provided
- [ ] Did you communicate these changes with the affected teams?
- [ ] Did unit tests/integration tests/E2E tests get properly added/updated?
- [ ] Is this change production-ready including testing, feature toggles, configurations, and deployment considerations?

#### Frontend Checklist:

- [x] Is this responsive and mobile friendly?
- [ ] Did you follow the [ARIA specs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA)

#### Backend Checklist:

- [ ] Does the endpoint return the correct success/error format?

---

**Screenshots**  
income chart

---

**Test Coverage**  
_add relevant screenshots here_

### CODE REVIEW

**Comment by KristofaJosh** [comment]
Task linked: [CU-86bwmcmbc Chart Overview](https://app.clickup.com/t/86bwmcmbc)

File: apps/client/src/pages/Dashboard/chartAndUpcoming/Chart/CustomTootip.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import PropTypes, { InferProps } from 'prop-types'
+import styles from './chat.module.scss'
+import Indicator from './images/Legend Indicator.svg?react'
+import Beak from './images/Bottom Beak.svg?react'
+import { Text } from 'components'
+
+interface DataPoint {
+  value?: number
+}
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (KristofaJosh) [inline apps/client/src/pages/Dashboard/chartAndUpcoming/Chart/CustomTootip.tsx:11]: was there no way to type this?

File: apps/client/src/pages/Dashboard/chartAndUpcoming/Chart/CustomTootip.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  payload?: DataPoint[]
+}
+export const CustomTooltip: React.FC<CustomTooltipProps> = ({
+  active,
+  payload,
+}) => {
+  if (active && payload && payload.length) {
+    return (
+      <div className={styles.customTip}>
+        {/* <div className={styles.customTipDot}></div> */}
```
Comment (KristofaJosh) [inline apps/client/src/pages/Dashboard/chartAndUpcoming/Chart/CustomTootip.tsx:22]: needed?

**Review by KristofaJosh — APPROVED** [review]

**Review by Elyzsium — APPROVED** [review]

**Review by Jidekas — DISMISSED** [review]

File: apps/client/src/pages/Dashboard/chartAndUpcoming/Chart/CustomTootip.tsx (hunk 1 above)
Comment (pelzman) [reply apps/client/src/pages/Dashboard/chartAndUpcoming/Chart/CustomTootip.tsx:11]: There should definitely be a way, but hasn’t figure it out yet

File: apps/client/src/pages/Dashboard/chartAndUpcoming/Chart/CustomTootip.tsx (hunk 2 above)
Comment (pelzman) [reply apps/client/src/pages/Dashboard/chartAndUpcoming/Chart/CustomTootip.tsx:22]: Yes it’s needed , is how the library accepts the customization

Comment (KristofaJosh) [reply apps/client/src/pages/Dashboard/chartAndUpcoming/Chart/CustomTootip.tsx:22]: I am referring to the comment

**Pull request merged**